runtime/binding-mqtt-kafka/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/kafka/internal/stream/MqttKafkaSessionFactory.java

```diff
@@ -338,8 +340,10 @@ public MessageConsumer newStream(
             final long resolvedId = resolved.id;
             final String16FW sessionTopic = binding.sessionsTopic();
 
+            final MqttQoS publishQosMax = binding.publishQosMax().value() < this.publishQosMax.value() ?
+                binding.publishQosMax() : this.publishQosMax;
```

Comment:
Please add a `MqttQos.min(MqttQos that)` method that returns the `MqttQos` with minimum `.value()`, either `this` or `that`.

```suggestion
            final MqttQoS publishQosMax = this.publishQosMax.min(binding.publishQosMax());
```